compiler/modules/front/expr.bal

```diff
@@ -56,7 +56,8 @@ type ExprNarrowing record {|
     Binding binding;
     t:SemType ifTrue;
     t:SemType ifFalse;
-    bir:InsnRef testInsn;
+    bir:Result trueBasis;
+    bir:Result falseBasis;
     boolean negated = false;
```

Comment:
IIRC I had `negated` so that I could generated the right basis from the InsnRef.  If you put the basis in, then I think you may be able to negate just by swapping the true/false fields, which would avoid the need for the smaller helper functions that are in effect just dealing with `negated`.